lib/galaxy/schema/workflows.py

```diff
@@ -194,19 +195,21 @@ class StoredWorkflowDetailed(StoredWorkflowSummary):
     )
     steps: Dict[
         int,
-        Union[
-            InputDataStep,
-            InputDataCollectionStep,
-            InputParameterStep,
-            PauseStep,
-            ToolStep,
-            SubworkflowStep,
+        Annotated[
+            Union[
+                InputDataStep,
+                InputDataCollectionStep,
+                InputParameterStep,
+                PauseStep,
+                ToolStep,
+                SubworkflowStep,
+            ],
+            Field(discriminator="type"),
```

Comment:
That's a good catch, we should backport that!

Reply:
Absolutely, I'll open a separate PR as I am not sure when this one will be ready.